src/content/reference/react/use-client.md

```diff
@@ -23,37 +23,364 @@ canary: true
 
 ### `'use client'` {/*use-client*/}
 
-Add `'use client';` at the very top of a file to mark that the file (including any child components it uses) executes on the client, regardless of where it's imported.
+Add `'use client'` at the top of a file to mark the module and its transitive dependencies as client code.
 
-```js
+```js {1}
 'use client';
 
 import { useState } from 'react';
+import { formatDate } from './formatters';
+import Link from './link';
 
 export default function RichTextEditor(props) {
+  formatDate();
   // ...
+  return
+    <Link />
+    // ...
+}
 ```
 
-When a file marked `'use client'` is imported from a server component, [compatible bundlers](/learn/start-a-new-react-project#bleeding-edge-react-frameworks) will treat the import as the "cut-off point" between server-only code and client code. Components at or below this point in the module graph can use client-only React features like [`useState`](/reference/react/useState).
+When a file marked with `'use client'` is imported from a Server Component, [compatible bundlers](/learn/start-a-new-react-project#bleeding-edge-react-frameworks) will treat the module import as a boundary between server-run and client-run code.
+
+As a dependency of `RichTextEditor`,  `formatDate` and `Link` will also be evaluated on the client regardless if their modules contain a `'use client'` directive.
+
+<DeepDive>
+
+#### Client and server rendering with React Server Components {/*client-and-server-rendering-with-react-server-components*/}
+
+React's full-stack architecture vision is realized through the [React Server Components (RSC) specification](https://github.com/reactjs/rfcs/blob/main/text/0188-server-components.md).
+
+React Server Components lets you coordinate which components of your [render tree](/learn/understanding-your-ui-as-a-tree#the-render-tree) are rendered on the server or on the client. Server rendering is traditionally more performant and allows access to the back-end, while client rendering is necessary for adding interactivity into your components.
```

Comment:
"traditionally" may not be the right word choice here (as opposed to actually, in the present day?) Perhaps you meant "often" or "generally".